tests/text/helpers.py

```diff
@@ -0,0 +1,538 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pickle
+import sys
+from enum import Enum, unique
+from functools import partial
+from typing import Any, Callable, Dict, Sequence, Union
+
+import numpy as np
+import pytest
+import torch
+from torch import Tensor
+from torch.multiprocessing import Pool, set_start_method
+
+from tests.helpers.testers import NUM_PROCESSES, setup_ddp
+from torchmetrics import Metric
+
+try:
+    set_start_method("spawn")
+except RuntimeError:
+    pass
+
+
+@unique
+class INPUT_ORDER(Enum):
+    PREDS_FIRST = 1
+    TARGETS_FIRST = 2
```

Comment:
so some of the metrics have the `preds, target` order and others have `target, preds`?

Reply:
Yes. This is case with the packages being tested against and the same order is used in our package's implementation as well.